js/components/MagnetMap/index.js

```diff
@@ -6,23 +6,58 @@ import MagnetMapMarker from './MagnetMapMarker';
 import {
   View,
   StyleSheet,
+  InteractionManager,
 } from 'react-native';
 
 export default class MagnetMap extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      usePlaceHolder: true,
+    };
+  }
+
+  componentDidMount() {
+    InteractionManager.runAfterInteractions(() => {
+      this.setState({
+        usePlaceHolder: false,
+      });
+    });
+  }
+
   render() {
-    const { style, region, children } = this.props;
+    let map = this.renderMap();
+    const { style } = this.props;
     return (
       <View style={[styles.root, style]}>
-        <MapView
-          provider={MapView.PROVIDER_GOOGLE}
-          style={styles.map}
-          region={region}
-          customMapStyle={mapStyle}>
-          {children}
-        </MapView>
+        {map}
       </View>
     );
   }
+
+  renderMap() {
+    if (this.state.usePlaceHolder) {
+      return this.renderPlaceHolder();
```

Comment:
Right, i did that thinking that maybe in the future, we can have like a real placeholder (image, spinner, whatever).

But can let the simplified version and then expand to what we have now if necessary.